packages/ai-mcp-server/src/node/mcp-theia-server-impl.ts

```diff
@@ -0,0 +1,339 @@
+        app.all('/mcp', async (req, res) => {
+            await this.handleStreamableHttpRequest(req, res);
+        });
+
+        app.get('/', (_req, res) => {
+            res.json({
+                name: 'MCP Theia Server',
+                version: '1.0.0',
+                endpoints: {
+                    mcp: '/mcp'
+                },
+                transport: 'streamable-http'
+            });
+        });
+    }
+
+    protected async handleStreamableHttpRequest(req: express.Request, res: express.Response): Promise<void> {
+        if (!this.server) {
+            res.status(503).json({
+                jsonrpc: '2.0',
+                error: {
+                    code: -32000,
+                    message: 'MCP Server not initialized',
+                },
+                id: undefined,
+            });
+            return;
+        }
+
+        const sessionId = req.headers['mcp-session-id'] as string | undefined;
+        let transport: StreamableHTTPServerTransport;
+
+        try {
+            if (sessionId && this.httpTransports.has(sessionId)) {
+                transport = this.httpTransports.get(sessionId)!;
+            } else {
+                transport = new StreamableHTTPServerTransport({
+                    sessionIdGenerator: () => randomUUID(),
+                    onsessioninitialized: newSessionId => {
+                        this.httpTransports.set(newSessionId, transport);
+                    }
+                });
+
+                transport.onclose = () => {
+                    if (transport.sessionId) {
+                        this.httpTransports.delete(transport.sessionId);
+                    }
+                };
+
+                await this.server.connect(transport);
+            }
+
+            await transport.handleRequest(req, res, req.body);
+        } catch (error) {
+            this.logger.error('Error handling MCP Streamable HTTP request:', error);
+            if (!res.headersSent) {
+                res.status(500).json({
+                    jsonrpc: '2.0',
+                    error: {
+                        code: -32603,
+                        message: `Internal server error: ${error}`,
+                    },
+                    id: req.body?.id || undefined,
+                });
+            }
+        }
+    }
+
+    protected async registerContributions(): Promise<void> {
+        if (!this.server) {
+            throw new Error('Server not initialized');
+        }
+
+        await this.backendContributionManager.registerBackendContributions(this.server);
+        await this.registerFrontendContributions();
+    }
+
+    protected getConfigFromEnvironment(): MCPServerConfig {
+        const enabled = process.env[MCP_SERVER_CONFIG.ENV_ENABLED] === 'true';
+        const port = process.env[MCP_SERVER_CONFIG.ENV_PORT] ?
+            parseInt(process.env[MCP_SERVER_CONFIG.ENV_PORT]!, 10) :
+            MCP_SERVER_CONFIG.DEFAULT_PORT;
```

Comment:
These configurations should not only be settable via environment variables. Instead it would be nice / nicer if they could be configured via the application options, i.e. in the package.json of the application